test/com/google/enterprise/adaptor/ad/AdAdaptorTest.java

```diff
@@ -1775,6 +1779,9 @@ public boolean ranFullCrawl() {
     private boolean ranFullCrawl;
     private boolean ranIncrementalCrawl;
 
+    public MoreFakeAdaptor() throws Exception {
```

Comment:
Added for symmetry, or did it have a purpose at some point along the way?

Reply:
The default ctor for the superclass can throw NamingException.

Reply:
If it's just NamingException, let's say that rather than Exception. I'm OK with tests that throw Exception rather than have a long throws clause, but I prefer helpers to be more accurate.

Reply:
Done.